### PR TITLE
libgpiod: Added version 1.6.4 and 2.0.1

### PR DESCRIPTION
**Pull request by ashley-b**
Specify library name and version:  **libgpiod/1.6.4**
- Added version 1.6.4
- Added new major version 2.0.1 with breaking API change
- Fixed test_package to work with version 1 & 2

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libgpiod/all/conandata.yml [hunk 1]
```diff
@@ -2,3 +2,9 @@ sources:
   "1.6.3":
     url: "https://git.kernel.org/pub/scm/libs/libgpiod/libgpiod.git/snapshot/libgpiod-1.6.3.tar.gz"
     sha256: "eb446070be1444fd7d32d32bbca53c2f3bbb0a21193db86198cf6050b7a28441"
+  "1.6.4":
+    url: "https://git.kernel.org/pub/scm/libs/libgpiod/libgpiod.git/snapshot/libgpiod-1.6.4.tar.gz"
+    sha256: "829d4ac268df07853609d67cfc7f476e9aa736cb2a68a630be99e8fad197be0a"
+  "2.0.1":
+    url: "https://git.kernel.org/pub/scm/libs/libgpiod/libgpiod.git/snapshot/libgpiod-2.0.1.tar.gz"
+    sha256: "b6eda55356160a8e73906e3d48e959ef81296787d764975b10f257e9660668e9"
```
Comment (jwillikers) [inline recipes/libgpiod/all/conandata.yml:10]: Can you order these from newest version to oldest version?

File: recipes/libgpiod/config.yml [hunk 2]
```diff
@@ -1,3 +1,7 @@
 versions:
   "1.6.3":
     folder: all
+  "1.6.4":
+    folder: all
+  "2.0.1":
+    folder: all
```
Comment (jwillikers) [inline recipes/libgpiod/config.yml:7]: Same here.

**Review by jwillikers — CHANGES_REQUESTED** [review]

**Review by jwillikers — APPROVED** [review]
Thanks! LGTM!

**Comment by ashley-b** [comment]
Just a follow up of this PR.

Thanks

**Comment by ashley-b** [comment]
@jwillikers Just a follow up of this PR.

Thanks

**Review by jwillikers — APPROVED** [review]

**Comment by jwillikers** [comment]
> @jwillikers Just a follow up of this PR.
> 
> Thanks

I've approved, but a Conan team member still needs to provide approval by review.

@RubenRBS  could review or assign a Conan member to review? I noticed  that one wasn't assigned automatically.

**Review by uilianries — APPROVED** [review]
LGTM

**Review by ericLemanissier — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Pull request merged**